commit d5313bcb7e56cd949ca920bb0c741a1d1d4093cf
Author: Florian Weimer <fweimer@redhat.com>
Date:   Fri Sep 23 19:30:57 2022 +0200

    nss: Use shared prefix in IPv4 address in tst-reload1
    
    Otherwise, sorting based on the longest-matching prefix in
    getaddrinfo can reorder the addresses in ways the test does not
    expect, depending on the IPv4 address of the host.
    
    Reviewed-by: Siddhesh Poyarekar <siddhesh@sourceware.org>
    (cherry picked from commit c02e29a0ba47d636281e1a026444a1a0a254aa12)

diff --git a/nss/tst-reload1.c b/nss/tst-reload1.c
index 27a18ed9c37607bb..844cfcddc5e0f638 100644
--- a/nss/tst-reload1.c
+++ b/nss/tst-reload1.c
@@ -43,12 +43,12 @@ static struct passwd pwd_table_1[] = {
 
 static const char *hostaddr_5[] =
   {
-   "ABCD", "abcd", "1234", NULL
+   "ABCd", "ABCD", "ABC4", NULL
   };
 
 static const char *hostaddr_15[] =
   {
-   "4321", "ghij", NULL
+   "4321", "4322", NULL
   };
 
 static const char *hostaddr_25[] =
@@ -86,12 +86,12 @@ static const char *hostaddr_6[] =
 
 static const char *hostaddr_16[] =
   {
-   "7890", "a1b2", NULL
+   "7890", "7891", NULL
   };
 
 static const char *hostaddr_26[] =
   {
-   "qwer", "tyui", NULL
+   "qwer", "qweR", NULL
   };
 
 static struct hostent host_table_2[] = {
